server/controllers/auth/signup.js

```diff
@@ -1,14 +1,34 @@
-// modules 
+// modules
+const Joi = require('@hapi/joi');
 const bcrypt = require('bcryptjs');
 const uuid = require('uuid/v1');
 const jwt = require('jsonwebtoken');
 
 const database = require("../../database");
 
+const signupSchema = Joi.object({
+	email: Joi.string()
+		.email()
+    .min(4)
+    .max(50)
+    .required(),
+	password: Joi.string()
+		.min(6)
+    .max(72)
+    .required(),
+});
+
 exports.signup = (req, res, next) => {
-  const email = req.body.email;
-  // password cannot be longer than 72 characters
-  const password = req.body.password;
+
+	const valid = signupSchema.validate(req.body);
+
+	if (valid.error) {
+		res.status(422);
+		next(valid.error);
+		return false;
+	}
+
+	const { email, password } = valid.value;
```

Comment:
```suggestion
	const email = valid.value.email;
	const password = valid.value.password;
```

Can you expand it? For better readability.

Reply:
de-structuring is considered a cleaner way. These are only two values, but in case we have more, it would make code lengthy unnecessarily.

Reply:
🤔in that case... ok